### PR TITLE
Nc news 11

### PR DESCRIPTION
**Pull request by denton909**
Hi 
I have completed task 11. Please can it be reviewed
Thanks
Phil

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -396,6 +396,78 @@ describe("nc-news-10", () => {
...
+             })
+            test('400 Bad Request if a try to order data by something other than ASC or DESC', () => {
+                return request(app)
+                .get('/api/articles?sortBy=article_id&orderBy=shoe')
+                .expect(400)
+                .then(({body})=>{
+                 expect(body.msg).toEqual('Bad Request Not A Valid Order By Request')
+                })
+             })
+        })
```
[the remark this answers is not in the document]
Comment (denton909) [reply __tests__/app.test.js:468]: Done

File: model.js [hunk 2]
```diff
@@ -19,9 +19,20 @@ exports.selectArticleById = (id) => {
...
+exports.selectArticles = (searchByTopic, sortBy = 'created_at', orderBy = 'DESC') => {
+    const queryValues = []
+    let queryStr = `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id`
+    if(searchByTopic){
+        queryValues.push(searchByTopic)
+        queryStr += ` WHERE articles.topic = $1`  
+    }
+    queryStr += ` GROUP BY articles.article_id ORDER BY ${sortBy} ${orderBy}`    
+    return db.query(queryStr, queryValues).then(({rows}) => {
+       if(!rows.length){
```
[the remark this answers is not in the document]
Comment (denton909) [reply model.js:38]: Done

File: model.js [hunk 3]
```diff
@@ -34,6 +41,41 @@ exports.selectArticles = (searchByTopic, sortBy = 'created_at', orderBy = 'DESC'
...
+    //     queryStr += ` WHERE articles.topic = $1`  
+    // }
+    // queryStr += ` GROUP BY articles.article_id ORDER BY ${sortBy} ${orderBy}`    
+    // return db.query(queryStr, queryValues).then(({rows}) => {
+    //    if(!rows.length){
+    //     return Promise.reject({status: 404, msg: "Request Not Found"})
+    //    } else {
+    //        return rows
+    //    }
+    // })
```
Comment (Dutton1990) [inline model.js:78]: Don't forget to remove this!

**Review by Dutton1990 — APPROVED** [review]
Super job, this ticket is done. You can merge!

File: model.js (hunk 3 above)
Comment (denton909) [reply model.js:78]: Done

**Pull request merged**